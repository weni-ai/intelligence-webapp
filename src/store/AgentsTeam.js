import { defineStore } from 'pinia';
import { computed, reactive, ref } from 'vue';

import nexusaiAPI from '@/api/nexusaiAPI.js';
import { useAlertStore } from './Alert';
import agentIconService from '@/utils/agentIconService';

import i18n from '@/utils/plugins/i18n';

export const useAgentsTeamStore = defineStore('AgentsTeam', () => {
  const linkToCreateAgent = 'https://github.com/weni-ai/weni-cli';

  const alertStore = useAlertStore();

  const activeTeam = reactive({
    status: null,
    data: {
      manager: null,
      agents: [],
    },
  });

  const officialAgents = reactive({
    status: null,
    data: [],
  });

  const myAgents = reactive({
    status: null,
    data: [],
  });

  const allAgents = computed(() => {
    return {
      manager: activeTeam.data.manager,
      agents: [...officialAgents.data, ...myAgents.data],
    };
  });

  const isAgentsGalleryOpen = ref(false);

  async function loadActiveTeam() {
    try {
      activeTeam.status = 'loading';

      const { data } = await nexusaiAPI.router.agents_team.listActiveTeam();

      activeTeam.data = {
        manager: data.manager,
        agents: agentIconService.applyIconsToAgents(data.agents),
      };
      activeTeam.status = 'complete';
    } catch (error) {
      console.error('error', error);

      activeTeam.status = 'error';
    }
  }

  async function loadOfficialAgents({ search = '' } = {}) {
    try {
      officialAgents.status = 'loading';

      const { data } = await nexusaiAPI.router.agents_team.listOfficialAgents({
        search,
      });

      officialAgents.data = agentIconService.applyIconsToAgents(data);
      officialAgents.status = 'complete';
    } catch (error) {
      console.error('error', error);

      officialAgents.status = 'error';
    }
  }

  async function loadMyAgents({ search = '' } = {}) {
    try {
      myAgents.status = 'loading';

      const { data } = await nexusaiAPI.router.agents_team.listMyAgents({
        search,
      });

      myAgents.data = agentIconService.applyIconsToAgents(data);
      myAgents.status = 'complete';
    } catch (error) {
      console.error('error', error);

      myAgents.status = 'error';
    }
  }

  async function toggleAgentAssignment({ uuid, is_assigned }) {
    if (!uuid || typeof is_assigned !== 'boolean') {
      throw new Error('uuid and is_assigned are required');
    }

    try {
      const agent =
        officialAgents.data.find((agent) => agent.uuid === uuid) ||
        myAgents.data.find((agent) => agent.uuid === uuid);

      if (!agent) {
        throw new Error('Agent not found');
      }

      const { data } =
        await nexusaiAPI.router.agents_team.toggleAgentAssignment({
          agentUuid: agent?.uuid,
          is_assigned,
        });

      agent.assigned = data.assigned;

      if (is_assigned) {
        activeTeam.data.agents.push(agent);
      } else {
        activeTeam.data.agents = activeTeam.data.agents.filter(
          (agent) => agent.uuid !== uuid,
        );
      }

      alertStore.add({
        text: i18n.global.t(
          is_assigned
            ? 'router.agents_team.card.success_assign_alert'
            : 'router.agents_team.card.success_unassign_alert',
          {
            agent: agent.name,
          },
        ),
        type: 'success',
      });

      return {
        status: 'success',
      };
    } catch (error) {
      console.error('error', error);

      alertStore.add({
        text: i18n.global.t('router.agents_team.card.error_alert'),
        type: 'error',
      });

      return {
        status: 'error',
      };
    }
  }

  return {
    linkToCreateAgent,
    activeTeam,
    officialAgents,
    myAgents,
    allAgents,
    isAgentsGalleryOpen,
    loadActiveTeam,
    loadOfficialAgents,
    loadMyAgents,
    toggleAgentAssignment,
  };
});
