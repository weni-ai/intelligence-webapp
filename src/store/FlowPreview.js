import { defineStore } from 'pinia';
import { ref } from 'vue';
import { get, attempt } from 'lodash';

import useFlowPreview from '@/composables/useFlowPreview';

export const useFlowPreviewStore = defineStore('flowPreview', () => {
  const {
    preview,
    previewInit,
    previewStart,
    previewHasQuickReplies,
    previewResume,
  } = useFlowPreview();

  const messages = ref([]);
  const messageInput = ref('');

  function addMessage(message) {
    messages.value.push(message);
  }

  function clearMessages() {
    messages.value = [];
  }

  function setMessageInput(text) {
    messageInput.value = text;
  }

  function replaceMessage(toReplace, newMessages) {
    const index = messages.value.indexOf(toReplace);
    if (index !== -1) {
      messages.value.splice(
        index,
        1,
        ...(Array.isArray(newMessages) ? newMessages : [newMessages]),
      );
    }
  }

  function removeMessage(message) {
    const index = messages.value.indexOf(message);
    if (index !== -1) {
      messages.value.splice(index, 1);
    }
  }

  function treatAnswerResponse(
    answer,
    data,
    {
      onBroadcast,
      onFlowStart,
      onMediaUnavailable,
      onCancelled,
      fallbackMessage = '',
    },
  ) {
    if (data.type === 'broadcast') {
      handleBroadcastResponse(answer, data, fallbackMessage, onBroadcast);
    } else if (data.type === 'flowstart') {
      // Insert a flowstart message before the answer
      const answerIndex = messages.value.indexOf(answer);
      messages.value.splice(answerIndex, 0, {
        type: 'flowstart',
        name: data.name,
        question_uuid: null,
        feedback: {
          value: null,
          reason: null,
        },
      });

      if (onFlowStart) onFlowStart(answer, data);
    } else if (data.type === 'media_and_location_unavailable') {
      answer.status = 'loaded';
      answer.type = data.type;

      if (onMediaUnavailable) onMediaUnavailable(answer);
    } else if (data.type === 'cancelled') {
      answer.status = 'loaded';
      removeMessage(answer);

      if (onCancelled) onCancelled();
    }
  }

  function handleBroadcastResponse(answer, data, fallbackMessage, onBroadcast) {
    answer.status = 'loaded';

    const message = get(data, 'message', fallbackMessage);
    const sources = get(data, 'fonts', []);

    if (Array.isArray(message) && message.length > 0) {
      answer.response = message[0];
      answer.sources = sources;

      createAdditionalMessages(message.slice(1), answer.question_uuid, sources);
    } else {
      answer.response = message;
      answer.sources = sources;
    }

    if (onBroadcast) onBroadcast(answer);
  }

  function createAdditionalMessages(items, questionUuid, sources) {
    items.forEach((item) => {
      const additionalMessage = {
        type: 'answer',
        status: 'loaded',
        response: item,
        question_uuid: questionUuid,
        sources,
        feedback: {
          value: null,
          reason: null,
        },
      };
      addMessage(additionalMessage);
    });
  }

  return {
    preview,
    messages,
    messageInput,

    previewInit,
    previewStart,
    previewResume,
    previewHasQuickReplies,

    addMessage,
    clearMessages,
    setMessageInput,
    replaceMessage,
    removeMessage,
    treatAnswerResponse,
  };
});
