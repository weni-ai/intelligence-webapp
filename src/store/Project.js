import { defineStore } from 'pinia';
import { ref } from 'vue';

import nexusaiAPI from '@/api/nexusaiAPI';

export const useProjectStore = defineStore('Project', () => {
  const uuid = sessionStorage.getItem('projectUuid') || '';
  const isMultiAgents = ref(null);
  const details = ref({
    status: null,
  });

  async function updateIsMultiAgents(boolean) {
    isMultiAgents.value = boolean;
  }

  async function getProjectDetails() {
    details.value.status = 'loading';

    try {
      const data = await nexusaiAPI.router.tunings.projectDetails.read({
        projectUuid: uuid,
      });

      details.value = data;
      details.value.status = 'success';
    } catch (error) {
      console.error(error);
      details.value.status = 'error';
    }
  }

  return {
    uuid,
    isMultiAgents,
    updateIsMultiAgents,
    getProjectDetails,
    details,
  };
});
