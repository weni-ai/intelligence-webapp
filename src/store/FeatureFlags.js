import { ref } from 'vue';
import { defineStore } from 'pinia';
import { computed, inject, watch } from 'vue';

import { useProjectStore } from './Project';

import { gbKey } from '../utils/Growthbook';
import env from '@/utils/env';

export const useFeatureFlagsStore = defineStore('FeatureFlags', () => {
  const growthbook = inject(gbKey);

  const currentProjectUuid = computed(() => useProjectStore().uuid || '');

  const getListAtEnv = (key) => {
    return env(key)?.split(',') || [];
  };

  const isProjectEnabledForFlag = (flagKey) => {
    const projectList = getListAtEnv(flagKey);
    return projectList.includes(currentProjectUuid.value);
  };

  const upgradeToMultiAgents = ref(false);

  const flags = computed(() => ({
    agentsTeam: useProjectStore().isMultiAgents,
    upgradeToMultiAgents,
    supervisorExport: isProjectEnabledForFlag('FF_SUPERVISOR_EXPORT'),
    newSupervisor: growthbook?.isOn('new_supervisor'),
  }));

  watch(
    currentProjectUuid,
    (newProjectUuid) => {
      if (newProjectUuid && growthbook) {
        growthbook.setAttributes({
          ...growthbook.getAttributes(),
          weni_project: newProjectUuid,
        });
      }
    },
    { immediate: true },
  );

  function editUpgradeToMultiAgents(value) {
    upgradeToMultiAgents.value = value;
  }

  return {
    flags,
    editUpgradeToMultiAgents,
  };
});
