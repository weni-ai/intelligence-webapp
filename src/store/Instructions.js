import { reactive, ref, computed } from 'vue';
import { defineStore } from 'pinia';

import { useProjectStore } from './Project';
import { useAlertStore } from './Alert';

import nexusaiAPI from '@/api/nexusaiAPI';

import i18n from '@/utils/plugins/i18n';

function callAlert(type, alertText) {
  const alertStore = useAlertStore();
  alertStore.add({
    text: i18n.global.t(`agent_builder.instructions.${alertText}`),
    type,
  });
}

export const useInstructionsStore = defineStore('Instructions', () => {
  const projectUuid = computed(() => useProjectStore().uuid);

  const instructions = reactive({
    data: [],
    status: null,
  });

  const newInstruction = reactive({
    text: '',
    status: null,
  });

  const activeInstructionsListTab = ref('custom');

  async function addInstruction() {
    newInstruction.status = 'loading';
    activeInstructionsListTab.value = 'custom';

    try {
      const instructionResponse =
        await nexusaiAPI.agent_builder.instructions.addInstruction({
          projectUuid: projectUuid.value,
          instruction: newInstruction,
        });

      instructions.data.unshift({
        ...newInstruction,
        id: instructionResponse.id,
      });
      newInstruction.status = null;
      newInstruction.text = '';

      callAlert('success', 'new_instruction.success_alert');
    } catch (error) {
      newInstruction.status = 'error';
      callAlert('error', 'new_instruction.error_alert');
    }
  }

  async function loadInstructions() {
    instructions.status = 'loading';
    try {
      const response = await nexusaiAPI.agent_builder.instructions.list({
        projectUuid: projectUuid.value,
      });

      instructions.data = [...instructions.data, ...response];
      instructions.status = 'complete';
    } catch (error) {
      instructions.status = 'error';
    }
  }
  async function editInstruction(id, text) {
    const instruction = instructions.data.find(
      (instruction) => instruction.id === id,
    );

    if (!instruction) return;

    try {
      instruction.status = 'loading';
      await nexusaiAPI.agent_builder.instructions.edit({
        projectUuid: projectUuid.value,
        id,
        text,
      });
      instruction.text = text;
      instruction.status = 'complete';

      callAlert('success', 'edit_instruction.success_alert');
    } catch (error) {
      instruction.status = 'error';
      callAlert('error', 'edit_instruction.error_alert');
    }

    return { status: instruction.status };
  }

  async function removeInstruction(id) {
    const instruction = instructions.data.find(
      (instruction) => instruction.id === id,
    );
    if (!instruction) return;
    instruction.status = 'loading';

    try {
      await nexusaiAPI.agent_builder.instructions.delete({
        projectUuid: projectUuid.value,
        id,
      });
      instructions.data = instructions.data.filter(
        (instruction) => instruction.id !== id,
      );
      callAlert('default', 'remove_instruction.success_alert');
    } catch (error) {
      instruction.status = 'error';
      callAlert('error', 'remove_instruction.error_alert');
    }

    return { status: instruction.status };
  }

  return {
    instructions,
    newInstruction,
    activeInstructionsListTab,
    addInstruction,
    loadInstructions,
    editInstruction,
    removeInstruction,
  };
});
