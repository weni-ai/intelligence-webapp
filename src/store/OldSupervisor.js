import { defineStore } from 'pinia';
import { computed, reactive, ref } from 'vue';
import { format, parseISO, subDays } from 'date-fns';

import { useProjectStore } from './Project';
import { useAlertStore } from './Alert';
import { useRoute } from 'vue-router';

import supervisorApi from '@/api/nexus/OldSupervisor';

import { PerformanceAdapter } from '@/api/adapters/supervisor/performance';
import i18n from '@/utils/plugins/i18n';

export const useOldSupervisorStore = defineStore('OldSupervisor', () => {
  const projectUuid = computed(() => useProjectStore().uuid);
  const alertStore = useAlertStore();
  const route = useRoute();
  const last30Days = format(subDays(new Date(), 30), 'yyyy-MM-dd');
  const today = format(new Date(), 'yyyy-MM-dd');

  const forwardStats = reactive({
    status: null,
    data: {
      attendedByAgent: 0,
      forwardedHumanSupport: 0,
    },
  });

  const conversations = reactive({
    status: null,
    data: [],
  });

  const selectedConversation = ref(null);

  const filters = reactive({
    start: route.query.start || last30Days,
    end: route.query.end || today,
    search: route.query.search || '',
    type: route.query.type || '',
    conversationId: route.query.conversationId || '',
  });

  async function loadForwardStats() {
    forwardStats.status = 'loading';
    try {
      const response = await supervisorApi.conversations.forwardStats({
        projectUuid: projectUuid.value,
        start: format(parseISO(filters.start), 'dd-MM-yyyy'),
        end: format(parseISO(filters.end), 'dd-MM-yyyy'),
      });

      const adaptedData = PerformanceAdapter.fromApi(response);
      const total =
        adaptedData.attendedByAgent + adaptedData.forwardedHumanSupport;

      const calculatePercentage = (value) =>
        total === 0 ? 0 : Math.round((value / total) * 100);

      forwardStats.data = {
        attendedByAgent: calculatePercentage(adaptedData.attendedByAgent),
        forwardedHumanSupport: calculatePercentage(
          adaptedData.forwardedHumanSupport,
        ),
      };

      forwardStats.status = 'complete';
    } catch (error) {
      forwardStats.status = 'error';
    }
  }

  async function loadConversations(page = 1) {
    conversations.status = 'loading';

    const formatDateParam = (date) => format(parseISO(date), 'dd-MM-yyyy');

    try {
      const response = await supervisorApi.conversations.list({
        projectUuid: projectUuid.value,
        page,
        start: formatDateParam(filters.start),
        end: formatDateParam(filters.end),
        search: filters.search,
        type: filters.type,
      });

      conversations.status = 'complete';
      conversations.data = response;
    } catch (error) {
      conversations.status = 'error';
    }
  }

  async function loadSelectedConversationData({ next = false } = {}) {
    const conversation = selectedConversation.value;

    if (!conversation) return;
    if (conversation.data.status === 'loading') return;
    if (next && !conversation.data.next) return;

    try {
      selectedConversation.value.data.status = 'loading';

      const params = {
        projectUuid: projectUuid.value,
        start: selectedConversation.value.created_on,
        end: selectedConversation.value.end_on,
        urn: selectedConversation.value.urn,
        next: next ? selectedConversation.value.data.next : null,
      };

      const response = await supervisorApi.conversations.getById(params);

      const mergedResults = next
        ? [...response.results, ...selectedConversation.value.data.results]
        : response.results;

      selectedConversation.value.data = {
        ...selectedConversation.value.data,
        ...response,
        results: mergedResults,
        status: 'complete',
      };
    } catch (error) {
      console.error('Error loading conversation data:', error);

      if (selectedConversation.value?.data) {
        selectedConversation.value.data.status = 'error';
      }
    }
  }

  function selectConversation(conversationId) {
    if (!conversationId) {
      selectedConversation.value = null;
      filters.conversationId = '';
      return;
    }

    const conversation = conversations.data.results.find(
      (conversation) => conversation.id == conversationId,
    );

    selectedConversation.value = {
      ...conversation,
      data: {
        status: null,
      },
    };
    filters.conversationId = conversationId;
  }

  async function exportSupervisorData({ token }) {
    try {
      await supervisorApi.conversations.export({
        hideGenericErrorAlert: true,
        projectUuid: projectUuid.value,
        token,
      });

      alertStore.add({
        type: 'success',
        text: i18n.global.t('agent_builder.supervisor.export.success'),
      });
    } catch (error) {
      alertStore.add({
        type: 'error',
        text: i18n.global.t('agent_builder.supervisor.export.error'),
      });
    }
  }

  return {
    forwardStats,
    loadForwardStats,
    conversations,
    loadConversations,
    loadSelectedConversationData,
    selectConversation,
    selectedConversation,
    filters,
    exportSupervisorData,
  };
});
