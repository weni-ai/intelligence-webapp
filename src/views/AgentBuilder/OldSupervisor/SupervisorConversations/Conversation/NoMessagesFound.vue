<template>
  <section class="no-messages-found">
    <UnnnicIcon
      icon="search_off"
      scheme="neutral-dark"
      class="no-messages-found__icon"
      data-testid="no-messages-icon"
    />

    <UnnnicIntelligenceText
      tag="p"
      color="neutral-dark"
      family="secondary"
      size="body-gt"
      weight="bold"
      class="no-messages-found__title"
      data-testid="no-messages-title"
    >
      {{ $t('agent_builder.supervisor.no_messages_found.title') }}
    </UnnnicIntelligenceText>

    <UnnnicIntelligenceText
      tag="p"
      color="neutral-dark"
      family="secondary"
      size="body-gt"
      data-testid="no-messages-description"
    >
      <!-- eslint-disable-next-line vue/component-name-in-template-casing -->
      <i18n-t
        keypath="agent_builder.supervisor.no_messages_found.description"
        tag="p"
        data-testid="no-messages-description-i18n"
      >
        <template #studio_link>
          <a
            :href="`${env('CONNECT_URL')}/projects/${projectStore.uuid}/studio/contact`"
            class="no-messages-found__link"
            target="_blank"
            data-testid="studio-link"
          >
            {{ $t('agent_builder.supervisor.no_messages_found.studio_link') }}
          </a>
        </template>
      </i18n-t>
    </UnnnicIntelligenceText>
  </section>
</template>

<script setup>
import { useProjectStore } from '@/store/Project';
import env from '@/utils/env';

const projectStore = useProjectStore();
</script>

<style scoped lang="scss">
.no-messages-found {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  height: calc(100vh - ($unnnic-spacing-xgiant + $unnnic-spacing-ant));

  text-align: center;

  & &__icon {
    $icon-size: 3.5 * $unnnic-font-size;

    font-size: $icon-size;
  }

  &__title {
    margin: $unnnic-spacing-ant 0 $unnnic-spacing-nano 0;
  }

  &__link {
    color: $unnnic-color-neutral-dark;
    font-weight: $unnnic-font-weight-bold;
    text-decoration: underline;
    text-underline-offset: 3px;
    cursor: pointer;
  }
}
</style>
