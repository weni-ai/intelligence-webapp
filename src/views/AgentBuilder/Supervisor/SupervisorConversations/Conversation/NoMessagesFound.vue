<template>
  <section class="no-messages-found">
    <UnnnicIcon
      icon="sms"
      scheme="neutral-soft"
      filled
      class="no-messages-found__icon"
      data-testid="no-messages-icon"
    />

    <UnnnicIntelligenceText
      tag="p"
      color="neutral-cloudy"
      family="secondary"
      size="body-gt"
      class="no-messages-found__title"
      data-testid="no-messages-title"
    >
      {{ $t('agent_builder.supervisor.no_messages_found.title') }}
    </UnnnicIntelligenceText>

    <UnnnicIntelligenceText
      tag="p"
      color="neutral-clean"
      family="secondary"
      size="body-md"
      data-testid="no-messages-description"
    >
      <!-- eslint-disable-next-line vue/component-name-in-template-casing -->
      <i18n-t
        keypath="agent_builder.supervisor.no_messages_found.description"
        tag="p"
        data-testid="no-messages-description-i18n"
      >
        <template #studio_link>
          <a
            :href="`${env('CONNECT_URL')}/projects/${projectStore.uuid}/studio/contact`"
            class="no-messages-found__link"
            target="_blank"
            data-testid="studio-link"
          >
            {{ $t('agent_builder.supervisor.no_messages_found.studio_link') }}
          </a>
        </template>
      </i18n-t>
    </UnnnicIntelligenceText>
  </section>
</template>

<script setup>
import { useProjectStore } from '@/store/Project';
import env from '@/utils/env';

const projectStore = useProjectStore();
</script>

<style scoped lang="scss">
.no-messages-found {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: $unnnic-spacing-nano;

  height: 100%;

  text-align: center;

  & &__icon {
    $icon-size: 3.5 * $unnnic-font-size;

    font-size: $icon-size;
  }

  &__link {
    color: $unnnic-color-neutral-clean;
    text-decoration: underline;
    text-underline-offset: 3px;
    cursor: pointer;
  }
}
</style>
