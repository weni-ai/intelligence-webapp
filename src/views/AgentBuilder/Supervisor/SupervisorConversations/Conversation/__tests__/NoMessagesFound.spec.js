import { mount } from '@vue/test-utils';
import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
import { createTestingPinia } from '@pinia/testing';
import { useProjectStore } from '@/store/Project';

import NoMessagesFound from '../NoMessagesFound.vue';

vi.stubEnv('CONNECT_URL', 'https://connect.example.com');

vi.mock('@/store/Project', () => ({
  useProjectStore: vi.fn().mockReturnValue({
    uuid: 'test-project-uuid-123',
  }),
}));

describe('NoMessagesFound.vue', () => {
  let wrapper;

  const pinia = createTestingPinia({ stubActions: false });

  const icon = () => wrapper.findComponent('[data-testid="no-messages-icon"]');
  const title = () =>
    wrapper.findComponent('[data-testid="no-messages-title"]');
  const description = () =>
    wrapper.findComponent('[data-testid="no-messages-description"]');
  const descriptionI18n = () =>
    wrapper.findComponent('[data-testid="no-messages-description-i18n"]');
  const studioLink = () => wrapper.find('[data-testid="studio-link"]');

  const createWrapper = () => {
    return mount(NoMessagesFound, {
      global: {
        plugins: [pinia],
      },
    });
  };

  beforeEach(() => {
    wrapper = createWrapper();
  });

  afterEach(() => {
    wrapper.unmount();
    vi.clearAllMocks();
  });

  describe('Component rendering', () => {
    it('should match snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders correctly with all required elements', () => {
      expect(wrapper.exists()).toBe(true);
      expect(wrapper.find('.no-messages-found').exists()).toBe(true);
    });

    it('renders the search_off icon with correct scheme', () => {
      expect(icon().exists()).toBe(true);
      expect(icon().props('icon')).toBe('sms');
      expect(icon().props('scheme')).toBe('neutral-soft');
      expect(icon().classes()).toContain('no-messages-found__icon');
    });

    it('renders the title with correct translation key', () => {
      expect(title().props('tag')).toBe('p');
      expect(title().props('color')).toBe('neutral-cloudy');
      expect(title().props('family')).toBe('secondary');
      expect(title().props('size')).toBe('body-gt');
      expect(title().props('weight')).toBe('regular');
      expect(title().classes()).toContain('no-messages-found__title');
    });

    it('renders the description with i18n-t component', () => {
      expect(description().exists()).toBe(true);
      expect(descriptionI18n().exists()).toBe(true);
      expect(descriptionI18n().props('keypath')).toBe(
        'agent_builder.supervisor.no_messages_found.description',
      );
      expect(descriptionI18n().props('tag')).toBe('p');
    });

    it('renders the studio link with correct attributes', () => {
      expect(studioLink().exists()).toBe(true);
      expect(studioLink().attributes('target')).toBe('_blank');
      expect(studioLink().classes()).toContain('no-messages-found__link');
    });
  });

  describe('Link construction', () => {
    it('constructs the correct studio link URL using env and project store', () => {
      const expectedUrl =
        'https://connect.example.com/projects/test-project-uuid-123/studio/contact';

      expect(studioLink().attributes('href')).toBe(expectedUrl);
      expect(studioLink().exists()).toBe(true);
    });

    it('handles different project UUID values correctly', async () => {
      const expectedUrl =
        'https://connect.example.com/projects/different-project-uuid/studio/contact';

      vi.mocked(useProjectStore).mockImplementationOnce(() => ({
        uuid: 'different-project-uuid',
      }));
      wrapper = createWrapper();

      expect(studioLink().attributes('href')).toBe(expectedUrl);
    });
  });

  describe('Accessibility and external link behavior', () => {
    it('opens the studio link in a new tab', () => {
      expect(studioLink().attributes('target')).toBe('_blank');
    });
  });
});
