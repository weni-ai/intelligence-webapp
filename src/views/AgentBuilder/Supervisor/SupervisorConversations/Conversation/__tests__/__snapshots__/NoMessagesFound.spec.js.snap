// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NoMessagesFound.vue > Component rendering > should match snapshot 1`] = `
"<section data-v-f59e04f8="" class="no-messages-found"><span data-v-9846e6db="" data-v-f59e04f8="" class="material-symbols-rounded unnnic-icon-scheme--neutral-soft unnnic-icon-size--md material-symbols-rounded--filled no-messages-found__icon" data-testid="no-messages-icon" translate="no">sms</span>
  <p data-v-ec8da4ad="" data-v-f59e04f8="" class="unnnic-text unnnic-text--size-body-gt unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-cloudy no-messages-found__title no-messages-found__title" data-testid="no-messages-title">No messages found in this conversation</p>
  <p data-v-ec8da4ad="" data-v-f59e04f8="" class="unnnic-text unnnic-text--size-body-md unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-clean" data-testid="no-messages-description">
    <!-- eslint-disable-next-line vue/component-name-in-template-casing -->
  <p data-v-f59e04f8="" data-testid="no-messages-description-i18n">If the message was sent by human support or triggered an automatic action, you can find it in <a data-v-f59e04f8="" href="https://connect.example.com/projects/test-project-uuid-123/studio/contact" class="no-messages-found__link" target="_blank" data-testid="studio-link">Studio</a>.</p>
  </p>
</section>"
`;
